src/main/java/seedu/address/logic/commands/FacebookAddAllFriendsCommand.java

```diff
@@ -0,0 +1,178 @@
+package seedu.address.logic.commands;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import facebook4j.Facebook;
+import facebook4j.FacebookException;
+import facebook4j.Paging;
+import facebook4j.ResponseList;
+import facebook4j.TaggableFriend;
+import javafx.scene.web.WebEngine;
+import seedu.address.commons.core.EventsCenter;
+import seedu.address.commons.events.ui.NewResultAvailableEvent;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.ui.BrowserPanel;
+
+
+//@@author alexfoodw
+/**
+ * Adds all available friends from a personal Facebook account.
+ * Current Maximum friends is set at 30.
+ */
+
+public class FacebookAddAllFriendsCommand extends UndoableCommand {
+    public static final String COMMAND_WORD = "facebookaddallfriends";
+    public static final String COMMAND_ALIAS = "fbaddall";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": adds all available friends from a Facebook account. (Due to current runtime constraints, "
+            + "maximum friends that can be added is capped at 30.)\n"
+            + "Alias: " + COMMAND_ALIAS + "\n";
+    public static final String MESSAGE_FACEBOOK_ADD_ALL_FRIENDS_ERROR = "Error with Facebook Tagable Friends API call."
+            + "User may not be registered as 'Test User'";
+    public static final String MESSAGE_FACEBOOK_ADD_ALL_FRIENDS_PAGING_ERROR = "Error with getting next page";
+    public static final String MESSAGE_FACEBOOK_ADD_ALL_FRIENDS_SUCCESS = " valid friends added from Facebook!";
```

Comment:
Shouldn't since he's appending a variable in front~

```
EventsCenter.getInstance().post(new NewResultAvailableEvent(totalFriendsAdded
                + MESSAGE_FACEBOOK_ADD_ALL_FRIENDS_SUCCESS + " (From "
                + FacebookConnectCommand.getAuthenticatedUsername() + "'s account)", false));
```